backend/src/api/user/user.api.ts

```diff
@@ -1,54 +1,63 @@
 import { Router } from 'express';
-import { ApiPath, HttpCode, UsersApiPath } from '~/common/enums';
+import { ApiPath, HttpCode, UsersApiPath, UserType } from '~/common/enums';
 import { userService } from '~/services/services';
 
 const initUserApi = (apiRouter: Router): Router => {
   const userRouter = Router();
 
   apiRouter.use(ApiPath.USERS, userRouter);
 
-  userRouter.get(UsersApiPath.ROOT, async (_req, res) => {
+  userRouter.get(UsersApiPath.ROOT, async (_req, res, next) => {
     try {
-      const users = await userService.getAllUsers()
+      const users = await userService.getAllUsers();
       res.status(HttpCode.OK).json(users);
     } catch(error) {
-      res.status(HttpCode.NOT_FOUND).json(error);
+      next(error);
     }
   });
 
-  userRouter.get(UsersApiPath.$ID, async (_req, res) => {
+  userRouter.get(UsersApiPath.TYPE_$TYPE, async (req, res, next) => {
     try {
-      const user = await userService.getUserById(_req.params.id);
+      const users = await userService.getUsersByType(req.params.type as UserType);
+      res.status(HttpCode.OK).json(users);
+    } catch(error) {
+      next(error);
+    }
+  });
+
+  userRouter.get(UsersApiPath.$ID, async (req, res, next) => {
+    try {
+      const user = await userService.getUserById(req.params.id);
       res.status(HttpCode.OK).json(user);
     } catch(error) {
-      res.status(HttpCode.NOT_FOUND).json(error);
+      next(error);
     }
   });
 
-  userRouter.post(UsersApiPath.ROOT, async (_req, res) => {
+  userRouter.post(UsersApiPath.ROOT, async (req, res, next) => {
     try {
-      const user = await userService.createNewUser(_req.body);
+      const user = await userService.createNewUser(req.body);
       res.status(HttpCode.OK).json(user);
     } catch(error) {
-      res.status(HttpCode.BAD_REQUEST).json(error);
+      next(error);
     }
   });
 
-  userRouter.put(UsersApiPath.$ID, async (_req, res) => {
+  userRouter.put(UsersApiPath.$ID, async (req, res, next) => {
     try {
-      const user = await userService.updateUser(_req.params.id, _req.body);
+      const user = await userService.updateUser(req.params.id, req.body);
       res.status(HttpCode.OK).json(user);
     } catch(error) {
-      res.status(HttpCode.BAD_REQUEST).json(error);
+      next(error);
     }
   });
 
-  userRouter.delete(UsersApiPath.$ID, async (_req, res) => {
+  userRouter.delete(UsersApiPath.$ID, async (req, res, next) => {
     try {
-      await userService.deleteUser(_req.params.id);
-      res.status(HttpCode.NO_CONTENT).json();
+      const result = await userService.deleteUser(req.params.id);
+      res.status(HttpCode.NO_CONTENT).json({res: result});
```

Comment:
```suggestion
      const isDeleted = await userService.deleteUser(req.params.id);
      res.status(HttpCode.OK).json(isDeleted);
```
userService.deleteUser should return boolean